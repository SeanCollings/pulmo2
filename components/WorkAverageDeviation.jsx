import React, { useState, useEffect, useContext } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import { useTheme } from '../hooks/useTheme';
import {
  ProfileContext,
  WORK_AVERAGE_DEVIATION,
  WORK_AVE_DEV_CURRENT,
  WORK_AVE_DEV_IMPROVEMENT,
  WORK_AVE_DEV_UP,
  WORK_AVE_DEV_DOWN,
} from '../context/profile-context';

const getIconNameAndSize = (improvement) => {
  switch (improvement) {
    case WORK_AVE_DEV_UP:
      return ['menu-up', 24];
    case WORK_AVE_DEV_DOWN:
      return ['menu-down', 24];
    default:
      return ['minus', 17];
  }
};

const WorkAverageDeviation = (props) => {
  const theme = useTheme();
  const { profileContext } = useContext(ProfileContext);
  const {
    [WORK_AVE_DEV_CURRENT]: currentWorkAverageDeviation,
    [WORK_AVE_DEV_IMPROVEMENT]: workAverageImprovement,
  } = profileContext[WORK_AVERAGE_DEVIATION];
  const [workAverageDeviation, setWorkAverageDeviation] = useState(
    currentWorkAverageDeviation
  );

  useEffect(() => {
    setWorkAverageDeviation(currentWorkAverageDeviation);
  }, [currentWorkAverageDeviation]);

  const [iconName, iconSize] = getIconNameAndSize(workAverageImprovement);
  const opacityDarker = theme.DARK ? 0.63 : 0.8;

  return (
    <View style={{ ...styles.wadContainer }}>
      <Text
        style={{
          ...styles.wadTextStyle,
          color: theme.TEXT,
          opacity: opacityDarker,
        }}
      >
        {`Work Average deviation: ${workAverageDeviation.toFixed(2)}%`}
      </Text>
      <View style={{ justifyContent: 'center', paddingLeft: 5 }}>
        <MaterialCommunityIcons
          name={iconName}
          color={theme.BORDER}
          size={iconSize}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  wadContainer: {
    paddingVertical: 5,
    flexDirection: 'row',
  },
  wadTextStyle: {
    fontSize: 17,
    fontFamily: 'tit-light',
  },
});

export default WorkAverageDeviation;
