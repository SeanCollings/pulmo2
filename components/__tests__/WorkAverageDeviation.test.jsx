import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup } from 'react-native-testing-library';
import WorkAverageDeviation from '../WorkAverageDeviation';
import ProfileContextProvider, {
  WORK_AVE_DEV_CURRENT,
  WORK_AVE_DEV_TOTAL_ACTIVITIES,
  WORK_AVE_DEV_IMPROVEMENT,
  WORK_AVE_DEV_UP,
  WORK_AVE_DEV_DOWN,
  WORK_AVE_DEV_SAME,
  WORK_AVERAGE_DEVIATION,
} from '../../context/profile-context';

describe('WorkAverageDeviation - unit test', () => {
  let component;

  afterEach(cleanup);

  test('should render with improvement up', async () => {
    const profile = {
      [WORK_AVERAGE_DEVIATION]: {
        [WORK_AVE_DEV_CURRENT]: 17.64,
        [WORK_AVE_DEV_TOTAL_ACTIVITIES]: 5,
        [WORK_AVE_DEV_IMPROVEMENT]: WORK_AVE_DEV_UP,
      },
    };

    await act(async () => {
      component = create(
        <ProfileContextProvider profile={profile}>
          <WorkAverageDeviation />
        </ProfileContextProvider>
      );
    });
    expect(component).toMatchSnapshot();
  });

  test('should render with improvement down', async () => {
    const profile = {
      [WORK_AVERAGE_DEVIATION]: {
        [WORK_AVE_DEV_CURRENT]: 17.64,
        [WORK_AVE_DEV_TOTAL_ACTIVITIES]: 5,
        [WORK_AVE_DEV_IMPROVEMENT]: WORK_AVE_DEV_DOWN,
      },
    };
    await act(async () => {
      component = create(
        <ProfileContextProvider profile={profile}>
          <WorkAverageDeviation />
        </ProfileContextProvider>
      );
    });
    expect(component).toMatchSnapshot();
  });

  test('should render with improvement same', async () => {
    const profile = {
      [WORK_AVERAGE_DEVIATION]: {
        [WORK_AVE_DEV_CURRENT]: 17.64,
        [WORK_AVE_DEV_TOTAL_ACTIVITIES]: 5,
        [WORK_AVE_DEV_IMPROVEMENT]: WORK_AVE_DEV_SAME,
      },
    };
    await act(async () => {
      component = create(
        <ProfileContextProvider profile={profile}>
          <WorkAverageDeviation />
        </ProfileContextProvider>
      );
    });
    expect(component).toMatchSnapshot();
  });
});
