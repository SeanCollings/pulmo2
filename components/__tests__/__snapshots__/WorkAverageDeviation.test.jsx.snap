// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkAverageDeviation - unit test should render with improvement down 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#3e4b57",
        "fontFamily": "tit-light",
        "fontSize": 17,
        "opacity": 0.8,
      }
    }
  >
    Work Average deviation: 17.64%
  </Text>
  <View
    style={
      Object {
        "justifyContent": "center",
        "paddingLeft": 5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#CCC",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`WorkAverageDeviation - unit test should render with improvement same 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#3e4b57",
        "fontFamily": "tit-light",
        "fontSize": 17,
        "opacity": 0.8,
      }
    }
  >
    Work Average deviation: 17.64%
  </Text>
  <View
    style={
      Object {
        "justifyContent": "center",
        "paddingLeft": 5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#CCC",
            "fontSize": 17,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`WorkAverageDeviation - unit test should render with improvement up 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#3e4b57",
        "fontFamily": "tit-light",
        "fontSize": 17,
        "opacity": 0.8,
      }
    }
  >
    Work Average deviation: 17.64%
  </Text>
  <View
    style={
      Object {
        "justifyContent": "center",
        "paddingLeft": 5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#CCC",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
